modules/core/04-channel/types/upgrade.go

```diff
@@ -69,3 +71,22 @@ func (uf UpgradeFields) ValidateBasic() error {
 func (ut UpgradeTimeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
+
+// HasPassed returns true if the upgrade has passed the timeout height or timestamp
+func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, error) {
```

Comment:
I'm guessing `HasPassed` is used elsewhere? I'm currently unable to find references to it (though it is a jumbled forest of branches all around and I could just be missing one).

Reply:
it will be used in ChanTryUpgradeHandler, but please review that function [on this pr](https://github.com/cosmos/ibc-go/pull/3535/files) 🙈